lib/x/response_parser.rb

```diff
@@ -46,16 +45,21 @@ def parse(response:, array_class: nil, object_class: nil)
 
     private
 
+    def json?(response)
+      return false if response.instance_of?(Net::HTTPNoContent)
+
+      JSON.parse(response.body)
```

Comment:
This is better than before but it seems wasteful to parse the response body and throw away the result right before parsing JSON again in the `parse` method. I would move this check into the `parse` method, especially if it's a private method that you're not going to call from the `X::HTTPError` class.

Reply:
I agree, I changed to have a similar version than https://github.com/sferik/x-ruby/pull/38/commits/8d16719c8737159d003abf6eecc4d59fb6536491.